apps/api/src/tasks/pending-tasks.service.ts

```diff
@@ -86,11 +86,12 @@ export class PendingTasksService {
     };
 
     return {
+      relations: { dependsOn: true },
       where: [
-        { ...query, runnerId },
-        { ...query, runnerId: IsNull() },
+        { ...query, runnerId, dependsOn: { target: IsNull() } },
+        { ...query, runnerId: IsNull(), dependsOn: { target: IsNull() } },
       ],
-      order: { avgDuration: 'DESC' },
+      order: { priorityIndex: 'DESC', avgDuration: 'DESC' },
```

Comment:
its not logical to mix priorities driven by the client with a method to return the longest running task.